factory_linux.go

```diff
@@ -192,6 +192,18 @@ func (l *LinuxFactory) Type() string {
 	return "libcontainer"
 }
 
+func (l *LinuxFactory) Destroy() error {
+	ls, err := ioutil.ReadDir(l.Root)
+	if err != nil {
+		return newGenericError(err, SystemError)
+	}
+	if len(ls) != 0 {
+		return newGenericError(fmt.Errorf("%s is not empty", l.Root), RootIsNotEmpty)
+	}
+	syscall.Unmount(l.Root, syscall.MNT_DETACH)
```

Comment:
We may create .libcontainer_mount file after mounting root, and then umount root only if this file exists.